stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -638,14 +767,32 @@ public final class _TensorComputation {
 
   // NOTE: the following properties are intentionally not implemented as an enum
   // due to high churn, *please do not refactor for Swiftiness*.
+  private var stateTFE: TFEState?
   private var stateTF: TFState?
 
-  /// The thread to run tensor computation in. The global config flag
-  /// '_RuntimeConfig.usesSynchronousExecution' decides whether tensor
-  /// computation should be synchronous: if true, this property will always be
-  /// nil. Otherwise, this property is non-nil only when the tensor computation
-  /// is on-going.
-  private var pthread: pthread_t?
+  /// The threads to run tensor computation in. In eager mode, we use N threads
+  /// when the tensor computation involves N device functions. In non-eager
+  /// mode, we use a single thread (that's sufficient as these N device
+  /// functions can be sent to the same TF_SessionRun() call.)
+  ///
+  /// The global config flag '_RuntimeConfig.usesSynchronousExecution' decides
+  /// whether tensor computation should be synchronous: if true, this property
+  /// will always be nil. Otherwise, this property is non-nil only when the
+  /// tensor computation is on-going.
+  /// TODO: Remove the `usesSynchronousExecution`mode as it is very limiting
+  /// (e.g. does not support running multiple device functions).
+  private var pthreads: [pthread_t] = []
+
+  /// The data structure to pass into pthread creation API. 
+  class FuncHandle {
+    public let computation: _TensorComputation
+    public let threadIdx: Int
```

Comment:
`threadIdx` -> `threadIndex`

Reply:
Done, though I don't think this adds clarity, similar to the above thread.